include/seastar/json/json_elements.hh

```diff
@@ -367,6 +370,47 @@ std::function<future<>(output_stream<char>&&)> stream_range_as_array(Container v
     };
 }
 
+/*!
+ * \brief consume values from a coroutine generator \c gen and encode the items using the \c fun mapping function
+ * onto the \c out output_stream as a json array.
+ */
+template<typename T, typename Func>
+requires requires (T val, Func aa, output_stream<char> s) { { formatter::write(s, aa(val)) } -> std::same_as<future<>>; }
+future<> generate_array_element(output_stream<char>& out, coroutine::experimental::generator<T>& gen, Func fun) {
+    bool first = true;
+    co_await out.write("[");
+    while (auto val = co_await gen()) {
+        if (first) {
+            first = false;
+        } else {
+            co_await out.write(", ");
+        }
+        co_await formatter::write(out, fun(*val));
+    }
+    co_await out.write("]");
+ }
+
+/*!
+ * \brief consume values from a coroutine generator \c gen and return a serialize function for it as a json array.
+ *
```

Comment:
What's a serialize function? Is it an existing concept?

Reply:
But what's a serialize function?

Reply:
I agree the name is bad.
This is actually a view transformation function from the native type T to a jsonable object.